### PR TITLE
Add info about models that support BackgroundFields

### PR DESCRIPTION
**Pull request by navidcy**

### CODE REVIEW

File: docs/src/model_setup/background_fields.md [hunk 1]
```diff
@@ -36,6 +36,9 @@ velocity field.
 Other possible terms associated with the Coriolis force, buoyancy, turbulence closures,
 and surface waves acting on background fields are neglected.
 
+!!! compat "Model compatibility"
+    At the moment _only_ the [`NonhydrostaticModel`](@ref) supports `BackgroundFields`.
```
Comment (glwagner) [inline docs/src/model_setup/background_fields.md:40]:
```suggestion
    `BackgroundFields` are only supported by [`NonhydrostaticModel`](@ref).
```

Comment (glwagner) [reply docs/src/model_setup/background_fields.md:40]: I feel like saying "at the moment" implies there are plans to implement support, but I don't think there are...

Comment (navidcy) [reply docs/src/model_setup/background_fields.md]: True. No need to imply anything about the future. Talk about the present.

**Review by francispoulin — APPROVED** [review]
Good addition.

**Pull request merged**